frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedItems.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+
+import {
+  SortableContainer,
+  SortableElement,
+} from "metabase/components/sortable";
+
+import ColumnItem from "./ColumnItem";
+
+interface SortableColumnFunctions {
+  onRemove?: (item: any) => void;
+  onEdit?: (item: any) => void;
+  onClick?: (item: any) => void;
+  onAdd?: (item: any) => void;
+  onEnable?: (item: any) => void;
+  getItemName: (item: any) => string;
+}
+
+interface SortableColumnProps extends SortableColumnFunctions {
+  item: any;
+}
+
+const SortableColumn = SortableElement(
+  ({
+    item,
+    getItemName,
+    onEdit,
+    onRemove,
+    onClick,
+    onAdd,
+    onEnable,
+  }: SortableColumnProps) => (
+    <ColumnItem
+      title={getItemName(item)}
+      onEdit={onEdit ? () => onEdit(item) : null}
+      onRemove={onRemove && item.enabled ? () => onRemove(item) : null}
```

Comment:
This is kind of a weird way to do this, would love suggestions for something cleaner